cosmic-applet-notifications/src/main.rs

```diff
@@ -0,0 +1,189 @@
+use cosmic::applet::CosmicAppletHelper;
+use cosmic::iced::wayland::{
+    popup::{destroy_popup, get_popup},
+    SurfaceIdWrapper,
+};
+use cosmic::iced::{
+    executor,
+    widget::{button, column, horizontal_rule, row, text, Row, Space},
+    window, Alignment, Application, Color, Command, Length, Subscription,
+};
+
+use cosmic::iced_style::application::{self, Appearance};
+use cosmic::iced_style::svg;
+use cosmic::theme::{self, Svg};
+use cosmic::widget::icon;
+use cosmic::widget::toggler;
+use cosmic::Renderer;
+use cosmic::{Element, Theme};
+
+use std::process;
+
+pub fn main() -> cosmic::iced::Result {
+    let helper = CosmicAppletHelper::default();
+    Notifications::run(helper.window_settings())
+}
+
+#[derive(Default)]
+struct Notifications {
+    applet_helper: CosmicAppletHelper,
+    theme: Theme,
+    icon_name: String,
+    popup: Option<window::Id>,
+    id_ctr: u32,
+    do_not_disturb: bool,
+    notifications: Vec<Vec<String>>,
+}
+
+#[derive(Debug, Clone)]
+enum Message {
+    TogglePopup,
+    DoNotDisturb(bool),
+    Settings,
+    Ignore,
+}
+
+impl Application for Notifications {
+    type Message = Message;
+    type Theme = Theme;
+    type Executor = executor::Default;
+    type Flags = ();
+
+    fn new(_flags: ()) -> (Notifications, Command<Message>) {
+        (
+            Notifications {
+                icon_name: "notification-alert-symbolic".to_string(),
+                ..Default::default()
+            },
+            Command::none(),
+        )
+    }
+
+    fn title(&self) -> String {
+        String::from("Notifications")
+    }
+
+    fn theme(&self) -> Theme {
+        self.theme
+    }
+
+    fn close_requested(&self, _id: SurfaceIdWrapper) -> Self::Message {
+        Message::Ignore
+    }
+
+    fn style(&self) -> <Self::Theme as application::StyleSheet>::Style {
+        <Self::Theme as application::StyleSheet>::Style::Custom(|theme| Appearance {
+            background_color: Color::from_rgba(0.0, 0.0, 0.0, 0.0),
+            text_color: theme.cosmic().on_bg_color().into(),
+        })
+    }
+
+    fn subscription(&self) -> Subscription<Message> {
+        Subscription::none()
+    }
+
+    fn update(&mut self, message: Message) -> Command<Message> {
+        match message {
+            Message::TogglePopup => {
+                if let Some(p) = self.popup.take() {
+                    destroy_popup(p)
+                } else {
+                    self.id_ctr += 1;
+                    let new_id = window::Id::new(self.id_ctr);
+                    self.popup.replace(new_id);
+
+                    let popup_settings = self.applet_helper.get_popup_settings(
+                        window::Id::new(0),
+                        new_id,
+                        (400, 300),
+                        Some(60),
```

Comment:
what is the padding for here?

Reply:
No reason, the notifications applet basically does nothing now, so I dont think that matters now. Finishing the applet is my priority next week.